compiler/src/compile.rs

```diff
@@ -1840,9 +1847,7 @@ impl Compiler {
 
     // Assign current position the given label
     fn set_label(&mut self, label: Label) {
-        let position = self.current_code_object().instructions.len();
-        // assert!(label not in self.label_map)
-        self.current_code_object().label_map.insert(label, position);
+        self.output_stack.last_mut().unwrap().set_label(label)
```

Comment:
You might consider creating a helper function `get_output_stream(&mut self)` for this code: `self.output_stack.last_mut().unwrap()`